airflow/bin/cli.py

```diff
@@ -361,13 +362,87 @@ def run(args, dag=None):
 
     task = dag.get_task(task_id=args.task_id)
     ti = TaskInstance(task, args.execution_date)
-    ti.refresh_from_db()
-    if ti.current_state() == "failed":
-        # This behavior is always correct (failed tasks should not run), but is implemented to specifically handle the
-        # case where a worker goes down, the celery task gets re-queued, and the scheduler kills the airflow task
-        # without knowing to revoke the task from the celery queue.
+    ti.refresh_from_db(include_job_id=True)
+    from airflow.utils.state import State
```

Comment:
I'm afraid to ask -- why does this import have to be local?

Reply:
If this can be a module-level import, we can also have a module-level set for non-retryable states

Reply:
I'm pretty sure it doesn't - I put it there out of fear.  I will move it up to the top level and see if it breaks anything.